daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -385,25 +409,45 @@ data HttpJsonResponseBody a = HttpJsonResponseBody
   } deriving (Generic)
 instance A.FromJSON a => A.FromJSON (HttpJsonResponseBody a)
 
+data AllocatePartyRequest = AllocatePartyRequest
+  { identifierHint :: TL.Text
+  , displayName :: TL.Text
+  } deriving (Generic)
+instance A.ToJSON AllocatePartyRequest
+
 -- | Run a request against the HTTP JSON API.
-httpJsonRequest :: A.FromJSON a => LedgerArgs -> Method -> Path -> IO a
-httpJsonRequest LedgerArgs {sslConfigM,tokM,port,host} method path = do
+httpJsonRequest :: A.FromJSON a => LedgerArgs -> Method -> Path -> (Request -> Request) -> IO a
+httpJsonRequest args method path modify = do
+  req <- makeRequest args method path modify
+  resp <- runHttp httpJSON req
+  pure $ result resp
+
+httpBsRequest :: LedgerArgs -> Method -> Path -> (Request -> Request) -> IO BS.ByteString
+httpBsRequest args method path modify = do
+  req <- makeRequest args method path modify
+  runHttp httpBS req
+
+makeRequest :: LedgerArgs -> Method -> Path -> (Request -> Request) -> IO Request
```

Comment:
I must admit that passing around functions to modify requests feels a bit strange. But I don't know anything less strange given the design of the HTTP library... 🤷

Reply:
:shrug: couldn't come up with something else either...